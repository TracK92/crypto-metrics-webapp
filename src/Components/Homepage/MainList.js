import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux/es/exports';
import { fetchCryptoData, selectCryptoData } from '../../redux/cryptoData.redux';
import Home from './Home';

const MainList = () => {
  const dispatch = useDispatch();
  const crypto = useSelector(selectCryptoData);

  useEffect(() => {
    dispatch(fetchCryptoData());
  }, [dispatch]);

  return (
    <div>
      <ul className="main_list">
        {crypto.map((item) => (
          <Home
            key={item.id}
            id={item.id}
            item={item}
            name={item.name}
            symbol={item.symbol}
            rank={item.rank}
            logo={item.logo}
            price={item.price}
            price_date={item.price_date}
            price_timestamp={item.price_timestamp}
            circulating_supply={item.circulating_supply}
            max_supply={item.max_supply}
            market_cap={item.market_cap}
            num_exchanges={item.num_exchanges}
            num_pairs={item.num_pairs}
            num_pairs_unmapped={item.num_pairs_unmapped}
            first_candle={item.first_candle}
            first_trade={item.first_trade}
            first_order_book={item.first_order_book}
            rank_delta={item.rank_delta}
            high={item.high}
            high_timestamp={item.high_timestamp}
            one_day={item['1d']}
            seven_day={item['7d']}
          />
        ))}
      </ul>
    </div>
  );
};

export default MainList;
