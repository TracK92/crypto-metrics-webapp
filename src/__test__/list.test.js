import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import MainList from '../Components/HomePage/MainList';

describe('test MainList component', () => {
  test('MainList should be in the document', () => {
    render(
      <Provider store={store}>
        <Router><MainList /></Router>
      </Provider>,
    );
    const home = screen.getByRole('list');
    expect(home).toBeInTheDocument();
  });
  test('MainList should match snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router><MainList /></Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
