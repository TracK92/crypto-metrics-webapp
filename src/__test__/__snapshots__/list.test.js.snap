// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MainList component MainList should match snapshot 1`] = `
<div>
  <div
    className="search_bar"
  >
    <input
      className="search"
      onChange={[Function]}
      placeholder="Search"
      type="text"
    />
  </div>
  <ul
    className="main_list"
  />
</div>
`;
